pkg/util/informermanager/single-cluster-manager.go

```diff
@@ -60,6 +66,14 @@ func (s *singleClusterInformerManagerImpl) ForResource(resource schema.GroupVers
 	s.appendHandler(resource, handler)
 }
 
+func (s *singleClusterInformerManagerImpl) IsInformerSynced(resource schema.GroupVersionResource) bool {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	_, exist := s.syncedInformers[resource]
```

Comment:
The `exist`, second return value, only reflects the `resource` in the cache, not means `synced`.

Reply:
The `syncedInformers` now store all informers that is synced. If we want to check whether the informers have synced every time, the function `WaitForCacheSync`, whose return value mark the synced informers, should be called because the interface `dynamicinformer.DynamicSharedInformerFactory` does not have a `HasSynced` function to easily check whether the informer has been synced. That is to say, we should call `WaitForCacheSync` every time, that may be a expensive. However, if we store synced informers when the function `WaitForCacheSync` is called, the check can be easily done.